modules/nextflow/src/main/groovy/nextflow/extension/PublishOp.groovy

```diff
@@ -100,24 +90,22 @@ class PublishOp {
      * two sub-directories eg work-dir/xx/yyyyyy/etc
      *
      * @param path
-     * @return
      */
     protected Path getTaskDir(Path path) {
-        if( path==null )
+        if( path == null )
             return null
-        def result = getTaskDir0(path, session.workDir)
-        if( result == null )
-            result = getTaskDir0(path, session.bucketDir)
-        return result
+        return getTaskDir0(path, session.workDir.resolve('tmp'))
+            ?: getTaskDir0(path, session.workDir)
+            ?: getTaskDir0(path, session.bucketDir)
     }
```

Comment:
Normally, the `overwrite` publish option is set to `!task.cached`. Since we are inferring the task work directory here, we should be able to recover also whether the task was cached by querying the cache db.

However that could put a lot of pressure on the cache db. Might be able to alleviate with a second cache in the publish op of task dir -> cached true/false.

Or we can implement #4729 and overwrite based on stale-ness. We would want to have a "lenient" mode (compare file save) and "deep" mode (compare file checksums) as there is a trade-off between performance and accuracy.

Reply:
I'd ignore the `task.cached` attribute and just check if the file metadata changed has it's done for regular resume mode. Cfr https://github.com/nextflow-io/nextflow/blob/509f271e70bd99f0dd867ea07ecf9c7831342091/modules/nf-commons/src/main/nextflow/util/CacheHelper.java#L382-L382

Reply:
I had to do a bit of hacking on `CacheHelper` to make it work. It normally uses the absolute file name, but here we want to use relative names. It's a bit ugly, might be cleaner to make CacheHelper into a builder i.e. `new HashBuilder(opts).with(...).with(...).build()`